src/Microsoft.Extensions.DiagnosticAdapter/DiagnosticSourceAdapter.cs

```diff
@@ -96,7 +96,23 @@ public void Write(string diagnosticName, object parameters)
             if (!succeeded)
             {
                 var newAdapter = _methodAdapter.Adapt(subscription.MethodInfo, parameters.GetType());
-                succeeded = newAdapter(_listener.Target, parameters);
+                try
+                {
+                    succeeded = newAdapter(_listener.Target, parameters);
+                }
+                catch (InvalidOperationException ex)
```

Comment:
Instead of doing this, define a new `Exception` type. This is exactly the use case for a user-defined exception type.